xchange-bitstamp/src/main/java/org/knowm/xchange/bitstamp/service/BitstampTradeService.java

```diff
@@ -43,28 +53,28 @@ public OpenOrders getOpenOrders() throws IOException, BitstampException {
 
   @Override
   public OpenOrders getOpenOrders(OpenOrdersParams params) throws ExchangeException, IOException {
-    Collection<CurrencyPair> pairs = DefaultOpenOrdersParamCurrencyPair.getPairs(params, exchange);
+
     List<LimitOrder> limitOrders = new ArrayList<>();
-    for (CurrencyPair pair : pairs) {
-      BitstampOrder[] openOrders = getBitstampOpenOrders(pair);
-      for (BitstampOrder bitstampOrder : openOrders) {
-        OrderType orderType = bitstampOrder.getType() == 0 ? OrderType.BID : OrderType.ASK;
-        String id = Long.toString(bitstampOrder.getId());
-        BigDecimal price = bitstampOrder.getPrice();
-        limitOrders.add(
-            new LimitOrder(
-                orderType,
-                bitstampOrder.getAmount(),
-                pair,
-                id,
-                bitstampOrder.getDatetime(),
-                price,
-                null, // avgPrice
-                null, // cumAmount
-                null, // fee
-                Order.OrderStatus.NEW));
-      }
+
```

Comment:
can this `new ArrayList<>()` be created before the `getBitstampOpenOrder()` and using the `openOrders.length` to size the array?

Reply:
yeah, that works